src/utils/time.ts

```diff
@@ -53,57 +86,23 @@ export const secondsToDuration = (totalSeconds: number): Duration => {
 export const formatSecondsAsDuration = (seconds: number): string =>
   formatDuration(secondsToDuration(seconds));
 
-const formatMeasurementInString = (value: number, measureUnit: string) => {
-  const baseFormatted = `${value} ${measureUnit}`;
-
-  if (value === 0) return "";
-  if (value === 1) return baseFormatted;
-  if (value > 1) return `${baseFormatted}s`;
-};
-
-// @debt quality test this
 export const getTimeBeforeParty = (startUtcSeconds?: number) => {
   if (startUtcSeconds === undefined) return "???";
 
-  const secondsBeforeParty =
-    startUtcSeconds - Date.now() / ONE_SECOND_IN_MILLISECONDS;
-
-  if (secondsBeforeParty < 0) {
-    return 0;
-  }
+  const eventStartDate = fromUnixTime(startUtcSeconds);
+  const now = Date.now();
 
-  const numberOfCompleteDaysBeforeParty = Math.floor(
-    secondsBeforeParty / ONE_DAY_IN_SECONDS
-  );
-
-  const numberOfCompleteHours = Math.floor(
-    (secondsBeforeParty % ONE_DAY_IN_SECONDS) / ONE_HOUR_IN_SECONDS
-  );
+  if (isBefore(eventStartDate, now)) return 0;
 
-  const numberOfMinutes = Math.floor(
-    (secondsBeforeParty % ONE_HOUR_IN_SECONDS) / ONE_MINUTE_IN_SECONDS
-  );
-
-  const numberOfDaysInString = formatMeasurementInString(
-    numberOfCompleteDaysBeforeParty,
-    "day"
-  );
-  const numberOfHoursInString = formatMeasurementInString(
-    numberOfCompleteHours,
-    "hour"
+  return formatDuration(
+    intervalToDuration({
+      start: Date.now(),
+      end: eventStartDate,
+    }),
+    { format: ["days", "hours", "minutes"] }
```

Comment:
[NB] Looking at https://date-fns.org/v2.21.3/docs/formatDuration#examples, instead of arbitrarily restricting ourselves to days/hours/minutes, I think what we really want to format here is just the duration without seconds.

So I would just use https://date-fns.org/v2.21.3/docs/intervalToDuration, and then manually set the `seconds` to `0` before passing it into this function.

Then you can get rid of this 'format days/hours/minutes' part here, and it will work properly even if it's days or months till the event.